changelog.d/13855.bugfix

```diff
@@ -0,0 +1 @@
+Fix access token leak to logs from proxy agent.
```

Comment:
Do we know where the `ProxyAgent` was being used? Is there anything other than the case described at https://github.com/matrix-org/synapse/issues/13010 which is worth highlighting?

Would be nice to have in the changelog, but don't stress if this isn't obvious.

Reply:
Looks like it's used all over but I guess that makes sense for the purpose of the proxy. Feel free to suggest better language here.

`SimpleHttpClient`
 - `ApplicationServiceApi`
 - `get_simple_http_client()`
 - `get_proxied_blacklisted_http_client()`
 
`MatrixFederationHttpClient`
 - `get_federation_http_client()`

`MatrixHostnameEndpoint`